src/resources/views/crud/inc/form_fields_script.blade.php

```diff
@@ -0,0 +1,183 @@
+
+            // Validate that the wrapper has been found
+            if (this.wrapper.length === 0) {
+                console.error(`CrudField error! Could not select WRAPPER for "${this.name}"`);
+            }
+
+            // Validate that the field has been found
+            if(this.$input.length === 0) {
+                console.error(`CrudField error! Could not select INPUT for "${this.name}"`);
+            }
+
+            this.input = this.$input[0];
+            this.type = this.wrapper.attr('bp-field-type');
+
+            return this;
+
+        }
+
+        get mainInput() {
+            let input = this.wrapper.find('[bp-field-main-input]').first();
+
+            // if a bp-field-main-input has been specified by developer, that's it, use that one
+            if (input.length !== 0) {
+                return input;
+            }
+
+            // otherwise, try to find the input using other selectors
+            if (this.$input.length === 0) {
+                // try searching for the field with the corresponding bp-field-name
+                input = this.wrapper.find('input[bp-field-name="'+this.name+'"], textarea[bp-field-name="'+this.name+'"], select[bp-field-name="'+this.name+'"], select[bp-field-name="'+this.name+'[]"]').first();
+
+                // if not input found yet, just get the first input in that wrapper
+                if (input.length === 0) {
+                    input = this.wrapper.find('input, textarea, select').first();
+                }
+
+                return input;
+            }
+
+            return this.$input;
+
+        }
+
+        get value() {
+            return this.$input.val();
+        }
+
+        get inputWrapper() {
+            let wrapper = this.$input.closest('[bp-field-wrapper]');
+            if (wrapper.length === 0) {
+                wrapper = $(`[bp-field-name="${this.name}"][bp-field-wrapper]`).first();
+            }
+            return wrapper;
+        }
+
+        onChange(closure) {
+            const bindedClosure = closure.bind(this);
+            const fieldChanged = (event, values) => bindedClosure(this, event, values);
+
+            if(this.isSubfield) {
+                window.crud.subfieldsCallbacks = window.crud.subfieldsCallbacks ?? [];
+                window.crud.subfieldsCallbacks[this.subfieldHolder] = window.crud.subfieldsCallbacks[this.subfieldHolder] ?? [];
+                if(!window.crud.subfieldsCallbacks[this.subfieldHolder].some( callbacks => callbacks['fieldName'] === this.name )) {
+                    window.crud.subfieldsCallbacks[this.subfieldHolder].push({closure: closure, field: this});
+                }
+                return this;
+            }
+
+            if(['INPUT', 'TEXTAREA'].includes(this.input?.nodeName)) {
+                this.input?.addEventListener('input', fieldChanged, false);
+            }
+            this.$input.change(fieldChanged);
+
+            return this;
+        }
+
```

Comment:
I don't know why we didn't think of this before, but... `.focus()`... I expect people will need to intentionally set the focus on a certain field... after something is done 😔

Reply:
My only concern about focus is that it needs to be on the current visible elements, it will break if you try to focus on an element in a different tab. 
And then we will be talking about `focus('tab.field')` and adding support to change to the `tab` and focus the `field`. 

Doable ... if it's really something that we need.